code/obj/machinery/plantpot.dm

```diff
@@ -138,15 +140,42 @@
 
 		frequency.post_signal(src, signal)
 
+	proc/update_water_level() //checks reagent contents of the pot, then returns the cuurent water level
+		var/current_total_volume = (src.reagents ? src.reagents.total_volume : 0)
+		var/current_water_level = (src.reagents ? src.reagents.get_reagent_amount("water") : 0)
+		switch(current_water_level)
+			if(0) current_water_level = 1
+			if(1 to 40) current_water_level = 2
+			if(41 to 100) current_water_level = 3
+			if(101 to 200) current_water_level = 4
+			if(201 to INFINITY) current_water_level = 5
+		if(current_water_level != src.water_level)
+			src.water_level = current_water_level
+		if(!current) // Make sure we don't have a plant in the tray first (layering stuffs)
+			switch(current_total_volume)
+				if(0) current_total_volume = 1
+				if(1 to 40) current_total_volume = 2
+				if(41 to 100) current_total_volume = 3
+				if(101 to 200) current_total_volume = 4
+				if(201 to INFINITY) current_total_volume = 5
```

Comment:
```suggestion
		switch(current_water_level)
			if(0 to 0) current_water_level = 1
			if(0 to 40) current_water_level = 2
			if(40 to 100) current_water_level = 3
			if(100 to 200) current_water_level = 4
			if(200 to INFINITY) current_water_level = 5
		if(current_water_level != src.water_level)
			src.water_level = current_water_level
		if(!current) // Make sure we don't have a plant in the tray first (layering stuffs)
			switch(current_total_volume)
				if(0 to 0) current_total_volume = 1
				if(0 to 40) current_total_volume = 2
				if(40 to 100) current_total_volume = 3
				if(100 to 200) current_total_volume = 4
				if(200 to INFINITY) current_total_volume = 5
```

Reply:
Yes, I know that `0 to 0` looks dumb but you need to do that. (You can just press the button on the suggestion above to incorporate it automatically into your branch!)